### PR TITLE
83782874 lbnl thermostatoffsetfault

### PR DESCRIPTION
**Pull request by rongpengzhang**
HumidistatOffsetFault: 
Distinguish two types of HumidistatOffset: (1) ThermostatOffset Independent (2) ThermostatOffset Dependent

### CODE REVIEW

**Comment by Myoldmopar** [comment]
This is only changing the brand new object, right?  No transition issues?

**Comment by rongpengzhang** [comment]
Yes, it only changes the object FaultModel:ThermostatOffset.

On Tue, Mar 10, 2015 at 9:34 AM, Edwin Lee <notifications@github.com> wrote:

> This is only changing the brand new object, right? No transition issues?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/NREL/EnergyPlus/pull/4782#issuecomment-78092314>.
>



-- 
Rongpeng Zhang, PhD
Building Technology and Urban Systems Division
Lawrence Berkeley National Laboratory, Berkeley, CA 94720

File: idd/Energy+.idd.in [hunk 1]
```diff
@@ -74805,24 +74807,40 @@ FaultModel:ThermostatOffset,
...
        \default 5
        \units percent
-   A4; \field Humidistat Name
+   A5, \field Related Thermostat Offset Fault Name
+       \note Enter the name of a FaultModel:ThermostatOffset object
+       \note Required field for Type ThermostatOffsetDependent
+       \type object-list
+       \object-list ThermostatOffsetFaults   
+   A6; \field Humidistat Name
        \begin-extensible
```
Comment (mjwitte) [inline idd/Energy+.idd.in]: @rongpengzhang Is this object extensible?  If so, there should be a line at the top of the object indicating \extensible:1   If not, this line should be deleted.

Comment (mjwitte) [inline idd/Energy+.idd.in]: And if it is not extensible, I'd put the humidistat name near the top of the object.

Comment (rongpengzhang) [inline idd/Energy+.idd.in]: @mjwitte 
No, it is not extensible. 
Yes, I think it's a better design to put the "humidistat name" near the top of the object, maybe just after the "fault name".
Thanks for the advice!

Comment (rongpengzhang) [inline idd/Energy+.idd.in]: @Myoldmopar  
I will update the iDD, cc codes, example files, and documentations to address Mike's comments. Please wait a while before merging this branch.

**Comment by rongpengzhang** [comment]
@Myoldmopar All green now :)

**Comment by rongpengzhang** [comment]
@Myoldmopar  Can we merge this branch into Develop? All tests are passed, and no transition issues.

File: idd/Energy+.idd.in [hunk 2]
```diff
@@ -74782,52 +74782,68 @@ FaultModel:EnthalpySensorOffset:ReturnAir,
...
        \type object-list
        \object-list ScheduleNames
-   N1, \field Reference Thermostat Offset
+   N1; \field Reference Thermostat Offset
        \required-field
        \type real
        \minimum> -10
        \maximum< 10
        \default 2
        \units C
```
Comment (mjwitte) [inline idd/Energy+.idd.in]: @rongpengzhang Sorry I did not notice this before.  units here should be deltaC

File: testfiles/Fault_HumidistatOffset_Supermarket.idf [hunk 3]
```diff
@@ -2431,8 +2431,10 @@
 	
   FaultModel:HumidistatOffset, 
     Humi_Offset_SalesFloor, !- Name
+    SalesFloor Humidistat,  !- Humidistat Name
+    ThermostatOffsetIndependent, !- Humidistat Offset Type
     ,  				  		!- Availability Schedule Name
     , 				  		!- Severity Schedule Name
     5,        	 	  		!- Reference Humidistat Offset
```
Comment (mjwitte) [inline testfiles/Fault_HumidistatOffset_Supermarket.idf]: @rongpengzhang Lines like this should have the units in the end-line comment
5,     !- Reference Humidistat Offset {percent}
Likewise for the thermostat offsets in other test files.
@Myoldmopar LKL used to run the example files through transition (I think) in a mode that "prettied" them up and fixed things like this automatically. Have you ever done that?

File: testfiles/Fault_ThermostatOffset_RefBldgMediumOfficeNew2004_Chicago.idf [hunk 4]
```diff
@@ -7913,106 +7913,106 @@
...
+    1.0;                     !- Reference Thermostat Offset
 	
   FaultModel:ThermostatOffset,
     Ther_Offset_Zn34,		 !- Name
+    Perimeter_top_ZN_4 Thermostat, 		 !- Thermostat Name
     ,                        !- Availability Schedule Name
     ,                        !- Severity Schedule Name
-    1.0,       				 !- Reference Thermostat Offset
-    Perimeter_top_ZN_4 Thermostat; 		 !- Thermostat Name
+    1.0;                     !- Reference Thermostat Offset
```
Comment (mjwitte) [inline testfiles/Fault_ThermostatOffset_RefBldgMediumOfficeNew2004_Chicago.idf]: @Myoldmopar Is this missing newline a problem?

File: idd/Energy+.idd.in (hunk 2 above)
Comment (rongpengzhang) [inline idd/Energy+.idd.in]: @mjwitte Thanks a lot for the check! I've corrected it.

File: testfiles/Fault_ThermostatOffset_RefBldgMediumOfficeNew2004_Chicago.idf (hunk 4 above)
Comment (rongpengzhang) [inline testfiles/Fault_ThermostatOffset_RefBldgMediumOfficeNew2004_Chicago.idf]: @mjwitte Do you mean the blank "Availability Schedule Name" and "Severity Schedule Name"? If they are blank, the schedule has values of 1.0 for all time periods.

Comment (mjwitte) [inline testfiles/Fault_ThermostatOffset_RefBldgMediumOfficeNew2004_Chicago.idf]: @rongpengzhang No, when I view this diff on github, line 8018 has a red 0 and an arrow and if you hover over that, it says "No newline at end of file".  I've ever seen that before.  Maybe it's normal.  @Myoldmopar ?

File: testfiles/Fault_HumidistatOffset_Supermarket.idf (hunk 3 above)
Comment (rongpengzhang) [inline testfiles/Fault_HumidistatOffset_Supermarket.idf]: @mjwitte 
@Myoldmopar told me the IDF texts would be re-processed (e.g., notes would be aligned). I guess the unit can be added too. Anyway, I will add the units in the IDFs.

I used "tabs" to align all the new added texts, and they look aligned in Notpad++. But when I open them in Github, they look unaligned.

Comment (mjwitte) [inline testfiles/Fault_HumidistatOffset_Supermarket.idf]: @rongpengzhang OK. We avoid using tabs in the idfs, spaces are better.

File: testfiles/Fault_ThermostatOffset_RefBldgMediumOfficeNew2004_Chicago.idf (hunk 4 above)
Comment (rongpengzhang) [inline testfiles/Fault_ThermostatOffset_RefBldgMediumOfficeNew2004_Chicago.idf]: @mjwitte I see what you mean. I've no idea what it is. 
capture

I tried to clean this up by copying the several ending lines from other IDF files to this one, but it did not work. Hopefully it doesn't matter. @Myoldmopar ?

**Comment by rongpengzhang** [comment]
@Myoldmopar 1645 of 1649 tests passed for the previous commit, so I merged Develop branch into this branch to pass all the tests.

**Comment by Myoldmopar** [comment]
CI failures are irrelevant.  I'm going to give the robots a few minutes to start on the last merge I did, but this should drop in today.

**Comment by Myoldmopar** [comment]
@rongpengzhang OK, so my comment from [10 hours ago](https://github.com/NREL/EnergyPlus/pull/4782#issuecomment-83030169) never went anywhere since apparently I never pushed the merge from develop that I did.  Anyway, it would be out of date by now.  So I merged it in again and pushed just now.  CI should have fun tonight and we'll merge it in the morning.

**Comment by rongpengzhang** [comment]
Thanks @Myoldmopar . I think CI had lots of fun last night ~

**Pull request merged**